swifttool/client.py

```diff
@@ -42,47 +42,55 @@ def _setup_logger(level=logging.INFO):
 
 
 @parallel
-def _fab_copy_swift_directory(local_files, remote_dir):
-    put(local_files, remote_dir, mirror_local_mode=True)
-
-
-@parallel
 def _fab_start_swift_services():
     with hide('running', 'stdout', 'stderr'):
         sudo("swift-init start all", pty=False, shell=False)
 
 
-def bootstrap(args):
-    rc = 0
+def scaleup(args):
+    rc = _manage(args)
+
+
+def scaledown(args):
+    rc = _manage(args, False)
+
+
+def _manage(args, scaleup=True):
     if not os.path.exists(args.config):
         raise Exception("Could not find confguration file '%s'" % args.config)
+    capman_helper(args.iterations, args.config, args.meta, args.outdir,
+                  scaleup)
 
-    config = yaml.load(open(args.config, 'r'))
-    ringsdef = SwiftRingsDefinition(config)
-
-    build_script = ringsdef.generate_script(outdir=args.outdir,
-                                            meta=args.meta)
-    subprocess.call(build_script)
 
-    tempfiles = os.path.join(ringsdef.workspace, "*")
-    execute(_fab_copy_swift_directory, tempfiles, args.outdir,
-            hosts=ringsdef.nodes)
-    execute(_fab_start_swift_services, hosts=ringsdef.nodes)
+def bootstrap(args):
+    if not os.path.exists(args.config):
+        raise Exception("Could not find confguration file '%s'" % args.config)
+    config = yaml.load(open(args.config, 'r'))
```

Comment:
You should definitely be using the `open` context manager here and take advantage of it to automagically close the file stream.  
```
with open(args.config, 'r') as f:
    config = yaml.load(f)
```

And you'll probably want to do an explicit check on the config to make sure it's not None.